app/src/main/java/de/danoeh/antennapod/fragment/actions/FeedMultiSelectActionHandler.java

```diff
@@ -0,0 +1,189 @@
+package de.danoeh.antennapod.fragment.actions;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.util.Log;
+
+import androidx.annotation.PluralsRes;
+
+import com.google.android.material.snackbar.Snackbar;
+
+import org.greenrobot.eventbus.EventBus;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.List;
+import java.util.Locale;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.core.dialog.ConfirmationDialog;
+import de.danoeh.antennapod.core.event.settings.SpeedPresetChangedEvent;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.dialog.RemoveFeedDialog;
+import de.danoeh.antennapod.fragment.FeedSettingsFragment;
+import de.danoeh.antennapod.fragment.preferences.dialog.PreferenceListDialog;
+import de.danoeh.antennapod.fragment.preferences.dialog.PreferenceSwitchDialog;
+import de.danoeh.antennapod.model.feed.Feed;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedPreferences;
+
+import static de.danoeh.antennapod.model.feed.FeedPreferences.SPEED_USE_GLOBAL;
+
+public class FeedMultiSelectActionHandler {
+    private static final String TAG = "FeedSelectHandler";
+    private final MainActivity activity;
+    private final List<Feed> selectedItems;
+
+    public FeedMultiSelectActionHandler(MainActivity activity, List<Feed> selectedItems) {
+        this.activity = activity;
+        this.selectedItems = selectedItems;
+    }
+
+    public void handleAction(int id) {
+        if (id == R.id.remove_item) {
+            deleteChecked();
+        } else if (id == R.id.keep_updated) {
+            keepUpdatedChecked();
+        } else if (id == R.id.autodownload) {
+            autoDownloadChecked();
+        } else if (id == R.id.autoDeleteDownload) {
+            autoDeleteEpisodesChecked();
+        } else if (id == R.id.playback_speed) {
+            playbackSpeedChecked();
+        } else {
+            Log.e(TAG, "Unrecognized speed dial action item. Do nothing. id=" + id);
+        }
+    }
+
+    private void autoDownloadChecked() {
+        PreferenceSwitchDialog preferenceSwitchDialog = new PreferenceSwitchDialog(activity,activity.getString(R.string.auto_download_settings_label), activity.getString(R.string.auto_download_label));
+        preferenceSwitchDialog.setOnPreferenceChangedListener(new PreferenceSwitchDialog.OnPreferenceChangedListener() {
+            @Override
+            public void preferenceChanged(boolean enabled) {
+                    saveFeedPreferences(feedPreferences -> feedPreferences.setAutoDownload(enabled));
+                    ApplyToEpisodesDialog dialog = new ApplyToEpisodesDialog(activity, enabled);
+                    dialog.createNewDialog().show();
+            }
+        });
+        preferenceSwitchDialog.openDialog();
+    }
+
+    public static final float SPEED_USE_GLOBAL = -1;
```

Comment:
Please don't create a copy of a magic value

Reply:
What should I use instead?

Reply:
`FeedPreferences.SPEED_USE_GLOBAL`